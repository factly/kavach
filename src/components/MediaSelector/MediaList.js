import React from 'react';
import { List, Avatar, Badge, Space, Input } from 'antd';
import { useDispatch, useSelector } from 'react-redux';
import { getMedia } from '../../actions/media';
import deepEqual from 'deep-equal';

function MediaList({ onSelect, selected }) {
  const dispatch = useDispatch();

  const [filters, setFilters] = React.useState({
    page: 1,
    limit: 8,
  });

  const { media, total } = useSelector((state) => {
    const node = state.media.req.find((item) => {
      return deepEqual(item.query, filters);
    });

    if (node)
      return {
        media: node.data.map((element) => state.media.details[element]),
        total: node.total,
      };

    return { media: [], total: 0 };
  });

  React.useEffect(() => {
    fetchMedia();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [dispatch, filters]);

  const fetchMedia = () => {
    dispatch(getMedia(filters));
  };

  return (
    <Space direction={'vertical'}>
      <Input
        placeholder="Search Media"
        onChange={(e) =>
          e.target.value ? setFilters({ ...filters, q: e.target.value }) : setFilters(filters)
        }
      />
      <List
        grid={{
          gutter: 16,
          md: 4,
        }}
        pagination={{
          current: filters.page,
          total: total,
          pageSize: filters.limit,
          onChange: (pageNumber, pageSize) => {
            setFilters({ page: pageNumber, limit: pageSize });
          },
        }}
        dataSource={media}
        renderItem={(item) => (
          <List.Item>
            {selected && item.id === selected.id ? (
              <Badge dot>
                <Avatar
                  onClick={() => onSelect(null)}
                  shape="square"
                  size={174}
                  src={item.url?.proxy}
                />
              </Badge>
            ) : (
              <Avatar
                onClick={() => onSelect(item)}
                shape="square"
                size={174}
                src={item.url?.proxy}
              />
            )}
          </List.Item>
        )}
      />
    </Space>
  );
}

export default MediaList;
