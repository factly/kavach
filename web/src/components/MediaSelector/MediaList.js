import React from 'react';
import { List, Avatar, Space, Input } from 'antd';
import { useDispatch, useSelector } from 'react-redux';
import { getMedia } from '../../actions/media';
import deepEqual from 'deep-equal';
import { CheckCircleTwoTone } from '@ant-design/icons';

function MediaList({ onSelect, selected, onUnselect }) {
  const dispatch = useDispatch();

  const [filters, setFilters] = React.useState({
    page: 1,
    limit: 8,
  });

  const { media, total } = useSelector((state) => {
    const node = state.media.req.find((item) => {
      return deepEqual(item.query, filters);
    });

    if (node)
      return {
        media: node.data.map((element) => state.media.details[element]),
        total: node.total,
      };

    return { media: [], total: 0 };
  });

  React.useEffect(() => {
    fetchMedia();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [dispatch, filters]);

  const fetchMedia = () => {
    dispatch(getMedia(filters));
  };

  return (
    <Space direction={'vertical'}>
      <Input
        placeholder="Search Media"
        onChange={(e) =>
          e.target.value ? setFilters({ ...filters, q: e.target.value }) : setFilters(filters)
        }
      />
      <List
        grid={{
          gutter: 16,
          md: 4,
        }}
        pagination={{
          current: filters.page,
          total: total,
          pageSize: filters.limit,
          onChange: (pageNumber, pageSize) => {
            setFilters({ page: pageNumber, limit: pageSize });
          },
        }}
        dataSource={media}
        renderItem={(item) => (
          <List.Item>
            {selected && item.id === selected.id ? (
              <div style={{ position: 'relative' }}>
                <Avatar
                  onClick={() => {
                    onSelect(null);
                    onUnselect();
                  }}
                  shape="square"
                  size={174}
                  src={(window.REACT_APP_ENABLE_IMGPROXY) ? item.url?.proxy : item.url?.raw}
                  style={{ opacity: '0.7', padding: '0.5rem', border: '2px solid #1890ff' }}
                />
                <CheckCircleTwoTone
                  twoToneColor="#52c41a"
                  style={{ fontSize: '2.5rem', position: 'absolute', top: 8, right: 8 }}
                />
              </div>
            ) : (
              <Avatar
                onClick={() => onSelect(item)}
                shape="square"
                size={174}
                src={(window.REACT_APP_ENABLE_IMGPROXY) ? item.url?.proxy : item.url?.raw}
              />
            )}
          </List.Item>
        )}
      />
    </Space>
  );
}

export default MediaList;
