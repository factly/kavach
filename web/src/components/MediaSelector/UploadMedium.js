import React from 'react';
import UppyUploader from '../Uppy';
import { useDispatch } from 'react-redux';
import { addMedium } from '../../actions/media';

function UploadMedium() {
  const dispatch = useDispatch();
  const onUpload = (values) => {
    dispatch(addMedium(values));
  };
  return <UppyUploader onUpload={onUpload} />;
}

export default UploadMedium;
