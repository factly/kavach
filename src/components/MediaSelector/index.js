import React from 'react';
import { Modal, Button, Radio, Space } from 'antd';
import { useDispatch, useSelector } from 'react-redux';
import MediaUploader from './UploadMedium';
import MediaList from './MediaList';
import { getMedium } from '../../actions/media';
import ImagePlaceholder from '../ErrorsAndImage/PlaceholderImage';

function MediaSelector({ value = null, onChange }) {
  const [show, setShow] = React.useState(false);
  const [selected, setSelected] = React.useState(null);
  const [tab, setTab] = React.useState('list');
  const dispatch = useDispatch();

  const medium = useSelector((state) => {
    return state.media.details[value] || null;
  });

  const setValue = () => {
    value = null;
  };

  React.useEffect(() => {
    if (value) {
      dispatch(getMedium(value));
      setSelected(medium);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <>
      <Modal
        visible={show}
        onCancel={() => setShow(false)}
        closable={false}
        width={'800px'}
        footer={[
          <Button key="back" onClick={() => setShow(false)}>
            Return
          </Button>,
          <Button
            key="submit"
            type="primary"
            onClick={() => {
              setShow(false);
              selected ? onChange(selected.id) : onChange(null);
            }}
          >
            Ok
          </Button>,
        ]}
      >
        <Space direction="vertical">
          <Radio.Group buttonStyle="solid" value={tab} onChange={(e) => setTab(e.target.value)}>
            <Radio.Button value="list">List</Radio.Button>
            <Radio.Button value="upload">Upload</Radio.Button>
          </Radio.Group>
          {tab === 'list' ? (
            <MediaList onSelect={setSelected} selected={selected} onUnselect={setValue}/>
          ) : tab === 'upload' ? (
            <MediaUploader />
          ) : null}
        </Space>
      </Modal>
      <Space direction="vertical">
        {medium ? (<img src={medium.url?.proxy} alt={medium.alt_text} width="100%" /> 
        ) : ( 
          <ImagePlaceholder />
        )}
        <Button onClick={() => setShow(true)}> Select</Button>
      </Space>
    </>      
  );
}

export default MediaSelector;