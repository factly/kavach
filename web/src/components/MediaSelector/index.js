import React from 'react';
import { Modal, Button, Radio, Space } from 'antd';
import { useDispatch, useSelector } from 'react-redux';
import MediaUploader from './UploadMedium';
import MediaList from './MediaList';
import { getMedium } from '../../actions/media';
import ImagePlaceholder from '../ErrorsAndImage/PlaceholderImage';
import { DeleteOutlined } from '@ant-design/icons';

function MediaSelector({
  value = null,
  onChange,
  maxWidth,
  containerStyles = {},
  profile = false,
}) {
  const [show, setShow] = React.useState(false);
  const [selected, setSelected] = React.useState(null);
  const [tab, setTab] = React.useState('list');
  const dispatch = useDispatch();

  const medium = useSelector((state) => {
    return state.media.details[value] || null;
  });

  const setValue = () => {
    value = null;
  };
  if (!selected && value && medium) {
    setSelected(medium);
  }
  React.useEffect(() => {
    if (value) {
      dispatch(getMedium(value, profile));
      setSelected(medium);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [value]);

  const onUpload = (values, medium) => {
    value = medium.id;
    setSelected(medium);
  };

  return (
    <>
      <Modal
        visible={show}
        onCancel={() => setShow(false)}
        closable={false}
        width={'800px'}
        footer={[
          <Button key="back" onClick={() => setShow(false)}>
            Cancel
          </Button>,
          <Button
            key="submit"
            type="primary"
            onClick={() => {
              setShow(false);
              selected ? onChange(selected.id) : onChange(null);
            }}
          >
            Confirm
          </Button>,
        ]}
      >
        <Space direction="vertical">
          <Radio.Group buttonStyle="solid" value={tab} onChange={(e) => setTab(e.target.value)}>
            <Radio.Button value="list">List</Radio.Button>
            <Radio.Button value="upload">Upload</Radio.Button>
          </Radio.Group>
          {tab === 'list' ? (
            <MediaList
              onSelect={setSelected}
              selected={selected}
              onUnselect={setValue}
              profile={profile}
            />
          ) : tab === 'upload' ? (
            <MediaUploader onMediaUpload={onUpload} profile={profile} />
          ) : null}
        </Space>
      </Modal>
      <Space direction="vertical">
        <div
          style={{
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
            ...containerStyles,
          }}
        >
          <div style={{ position: 'relative' }}>
            <Button
              style={{
                background: 'transparent',
                borderStyle: 'dashed',
                height: 'auto',
                display: 'block',
              }}
              onClick={() => setShow(true)}
            >
              {medium ? (
                <img
                  src={window.REACT_APP_ENABLE_IMGPROXY ? medium.url?.proxy : medium.url?.raw}
                  alt={medium.alt_text}
                  width="100%"
                />
              ) : (
                <ImagePlaceholder maxWidth={maxWidth} />
              )}
            </Button>
            {medium && (
              <Button
                style={{ position: 'absolute', bottom: 0, left: 0, maxWidth: '52px' }}
                onClick={() => {
                  onChange(null);
                  setSelected(null);
                }}
              >
                <DeleteOutlined />
              </Button>
            )}
          </div>
        </div>
      </Space>
    </>
  );
}

export default MediaSelector;
