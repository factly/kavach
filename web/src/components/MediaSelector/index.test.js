import React from 'react';
import { Provider } from 'react-redux';
import { useDispatch } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Radio } from 'antd';

import '../../matchMedia.mock';
import MediaSelector from './index';
import { mount } from 'enzyme';
import { act } from '@testing-library/react';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));
jest.mock('../../actions/media', () => ({
  ...jest.requireActual('../../actions/media'),
  getMedium: jest.fn(),
  getMedia: jest.fn(),
}));
let state = {
  organisations: {
    ids: [1],
    details: { 1: { id: 1, name: 'organisation' } },
    loading: false,
    selected: 1,
  },
  media: {
    req: [
      {
        data: [1, 2],
        query: { page: 1, limit: 8 },
        total: 2,
      },
    ],
    details: {
      1: {
        id: 1,
        name: 'Medium -1',
        url: 'some-url',
        file_size: 'file_size',
        caption: 'caption',
        description: 'description',
      },
      2: {
        id: 2,
        name: 'Medium - 2',
        url: 'some-url',
        file_size: 'file_size',
        caption: 'caption',
        description: 'description',
      },
    },
    loading: false,
  },
};

describe('MediaSelector component', () => {
  let store;
  let mockedDispatch;

  describe('snapshot testing', () => {
    beforeEach(() => {
      store = mockStore(() => {});
      store.dispatch = jest.fn();
      mockedDispatch = jest.fn();
      useDispatch.mockReturnValue(mockedDispatch);
    });

    it('should match component with data', () => {
      store = mockStore(state);
      const tree = mount(
        <Provider store={store}>
          <MediaSelector value={1} />
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper;
    beforeEach(() => {
      store = mockStore(state);
      store.dispatch = jest.fn(() => ({}));
      mockedDispatch = jest.fn(() => Promise.resolve({}));
      useDispatch.mockReturnValue(mockedDispatch);
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('should handle open and close modal', () => {
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <MediaSelector />
          </Provider>,
        );
      });
      const selectButton = wrapper.find('Button').at(0);
      expect(selectButton.text()).toBe(' Select');
      selectButton.simulate('click');
      expect(wrapper.find('Modal').props().visible).toBe(true);
      const closeButton = wrapper.find('Button').at(0);
      expect(closeButton.text()).toBe('Return');
      closeButton.simulate('click');
      expect(wrapper.find('Modal').props().visible).toBe(false);
    });
    it('should select image', () => {
      const onChange = jest.fn();
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <MediaSelector value={1} onChange={onChange} />
          </Provider>,
        );
      });
      const selectButton = wrapper.find('Button').at(0);
      expect(selectButton.text()).toBe(' Select');
      selectButton.simulate('click');
      wrapper.find('Avatar').at(1).simulate('click');
      const okButton = wrapper.find('Button').at(1);
      expect(okButton.text()).toBe('Ok');
      okButton.simulate('click');
      expect(onChange).toHaveBeenCalledWith(2);
    });
    it('should handle Tab change and unselect image', () => {
      const onChange = jest.fn();
      process.env.REACT_APP_COMPANION_URL = 'http://127.0.0.1:3020';
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <MediaSelector value={1} onChange={onChange} />
          </Provider>,
        );
      });
      const selectButton = wrapper.find('Button').at(0);
      expect(selectButton.text()).toBe(' Select');
      selectButton.simulate('click');
      wrapper.find('Avatar').at(0).simulate('click');
      wrapper
        .find(Radio.Group)
        .at(0)
        .props()
        .onChange({ target: { value: 'upload' } });
      wrapper.find('Modal').props().onCancel();
      expect(onChange).not.toHaveBeenCalled();
    });
    it('should handle for profile', () => {
      const onChange = jest.fn();
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <MediaSelector onChange={onChange} profile={true} />
          </Provider>,
        );
      });
      const selectButton = wrapper.find('Button').at(0);
      expect(selectButton.text()).toBe(' Select');
      selectButton.simulate('click');
      wrapper
        .find(Radio.Group)
        .at(0)
        .props()
        .onChange({ target: { value: 'invalid' } });
      const okButton = wrapper.find('Button').at(1);
      expect(okButton.text()).toBe('Ok');
      okButton.simulate('click');
      expect(onChange).toHaveBeenCalledWith(null);
    });
  });
});
