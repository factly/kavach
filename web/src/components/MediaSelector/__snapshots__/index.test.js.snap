// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaSelector component snapshot testing should match component with data 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MediaSelector
    value={1}
  >
    <Modal
      closable={false}
      footer={
        Array [
          <ForwardRef(InternalButton)
            onClick={[Function]}
          >
            Cancel
          </ForwardRef(InternalButton)>,
          <ForwardRef(InternalButton)
            onClick={[Function]}
            type="primary"
          >
            Confirm
          </ForwardRef(InternalButton)>,
        ]
      }
      onCancel={[Function]}
      visible={false}
      width="800px"
    >
      <NoCompactStyle>
        <NoFormStyle
          override={true}
          status={true}
        >
          <Dialog
            closable={false}
            closeIcon={
              <span
                className="ant-modal-close-x"
              >
                <ForwardRef(CloseOutlined)
                  className="ant-modal-close-icon"
                />
              </span>
            }
            focusTriggerAfterClose={true}
            footer={
              Array [
                <ForwardRef(InternalButton)
                  onClick={[Function]}
                >
                  Cancel
                </ForwardRef(InternalButton)>,
                <ForwardRef(InternalButton)
                  onClick={[Function]}
                  type="primary"
                >
                  Confirm
                </ForwardRef(InternalButton)>,
              ]
            }
            maskTransitionName="ant-fade"
            onCancel={[Function]}
            onClose={[Function]}
            prefixCls="ant-modal"
            transitionName="ant-zoom"
            visible={false}
            width="800px"
            wrapClassName=""
          >
            <Portal
              autoDestroy={false}
              autoLock={false}
              open={false}
            />
          </Dialog>
        </NoFormStyle>
      </NoCompactStyle>
    </Modal>
    <Space
      direction="vertical"
    >
      <div
        className="ant-space ant-space-vertical"
        style={Object {}}
      >
        <Item
          className="ant-space-item"
          direction="vertical"
          index={0}
          key="ant-space-item-0"
          marginDirection="marginRight"
          wrap={false}
        >
          <div
            className="ant-space-item"
            style={Object {}}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <div
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <Button
                  onClick={[Function]}
                  style={
                    Object {
                      "background": "transparent",
                      "borderStyle": "dashed",
                      "display": "block",
                      "height": "auto",
                    }
                  }
                >
                  <Wave
                    disabled={false}
                  >
                    <button
                      className="ant-btn ant-btn-default"
                      disabled={false}
                      onClick={[Function]}
                      style={
                        Object {
                          "background": "transparent",
                          "borderStyle": "dashed",
                          "display": "block",
                          "height": "auto",
                        }
                      }
                      type="button"
                    >
                      <LoadingIcon
                        existIcon={false}
                        loading={false}
                        prefixCls="ant-btn"
                      >
                        <CSSMotion
                          motionName="ant-btn-loading-icon-motion"
                          onAppearActive={[Function]}
                          onAppearStart={[Function]}
                          onEnterActive={[Function]}
                          onEnterStart={[Function]}
                          onLeaveActive={[Function]}
                          onLeaveStart={[Function]}
                          removeOnLeave={true}
                          visible={false}
                        >
                          <DomWrapper />
                        </CSSMotion>
                      </LoadingIcon>
                      <img
                        key=".0"
                        width="100%"
                      />
                    </button>
                  </Wave>
                </Button>
                <Button
                  onClick={[Function]}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "maxWidth": "52px",
                      "position": "absolute",
                    }
                  }
                >
                  <Wave
                    disabled={false}
                  >
                    <button
                      className="ant-btn ant-btn-default"
                      disabled={false}
                      onClick={[Function]}
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "maxWidth": "52px",
                          "position": "absolute",
                        }
                      }
                      type="button"
                    >
                      <LoadingIcon
                        existIcon={false}
                        loading={false}
                        prefixCls="ant-btn"
                      >
                        <CSSMotion
                          motionName="ant-btn-loading-icon-motion"
                          onAppearActive={[Function]}
                          onAppearStart={[Function]}
                          onEnterActive={[Function]}
                          onEnterStart={[Function]}
                          onLeaveActive={[Function]}
                          onLeaveStart={[Function]}
                          removeOnLeave={true}
                          visible={false}
                        >
                          <DomWrapper />
                        </CSSMotion>
                      </LoadingIcon>
                      <ForwardRef(DeleteOutlined)
                        key=".0"
                      >
                        <AntdIcon
                          icon={
                            Object {
                              "icon": Object {
                                "attrs": Object {
                                  "focusable": "false",
                                  "viewBox": "64 64 896 896",
                                },
                                "children": Array [
                                  Object {
                                    "attrs": Object {
                                      "d": "M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z",
                                    },
                                    "tag": "path",
                                  },
                                ],
                                "tag": "svg",
                              },
                              "name": "delete",
                              "theme": "outlined",
                            }
                          }
                        >
                          <span
                            aria-label="delete"
                            className="anticon anticon-delete"
                            role="img"
                          >
                            <IconReact
                              icon={
                                Object {
                                  "icon": Object {
                                    "attrs": Object {
                                      "focusable": "false",
                                      "viewBox": "64 64 896 896",
                                    },
                                    "children": Array [
                                      Object {
                                        "attrs": Object {
                                          "d": "M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z",
                                        },
                                        "tag": "path",
                                      },
                                    ],
                                    "tag": "svg",
                                  },
                                  "name": "delete",
                                  "theme": "outlined",
                                }
                              }
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="delete"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                key="svg-delete"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                                  key="svg-delete-svg-0"
                                />
                              </svg>
                            </IconReact>
                          </span>
                        </AntdIcon>
                      </ForwardRef(DeleteOutlined)>
                    </button>
                  </Wave>
                </Button>
              </div>
            </div>
          </div>
        </Item>
      </div>
    </Space>
  </MediaSelector>
</Provider>
`;
